pandas/core/dtypes/cast.py

```diff
@@ -468,7 +487,7 @@ def maybe_casted_values(index, codes=None):
 
     # if we have the codes, extract the values with a mask
     if codes is not None:
-        mask = codes == -1
+        mask: np.ndarray = codes == -1
```

Comment:
why was this added. mypy is not reporting any errors for me with this removed.

generally, try to avoid adding variable type annotations. should be inferred from expression.

Reply:
mistake, will revert

I had tried typing `index` and `codes` but there were a lot of errors - this was part of an attempted fix.